src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx

```diff
@@ -42,8 +42,8 @@ function OnboardingModalNavigator() {
         Navigation.isNavigationReady().then(() => {
             // Need to go back to previous route and then redirect to Concierge,
             // otherwise going back on Concierge will go to onboarding and then redirected to Concierge again
-            Navigation.goBack();
-            Report.navigateToConciergeChat();
+            Navigation.setShouldPopAllStateOnUP(true);
+            Navigation.goBack(ROUTES.HOME, true, true);
```

Comment:
On navigate to onboarding with deeplink it shows the not found page for a moment. 

https://github.com/user-attachments/assets/1d893d4a-9b51-4fc1-9c49-eb2f03d21ac8

I think we don't need any change in `OnboardingWelcomeVideo` component, we can just set navigation based on the smaller screen width like below.

```js
if(isSmallScreenWidth) {
  Navigation.setShouldPopAllStateOnUP(true);
  Navigation.goBack(ROUTES.HOME, true, true);
} else {
  Navigation.goBack();
}
```

Also, I believe for any purpose we need to do the same thing. That we will confirm with @mountiny .

Reply:
This feels like a real edge case, not sure if we should do much about it if it complicated the code further cc @filip-solecki for review

Reply:
I am suggesting to simplify it directly with the screen size 
- For small screen just pop up to the home
- For wider screen just use the goBack to close the modal and stay on the same screen

Reply:
I agree with @Pujan92